### ISSUE TITLE
False positive with local variable defined but may not be used

### ISSUE THREAD

**Issue by GShotwell** [r-lib/lintr#420]
Thanks for the great package, I'm running into a false positive using lintr along with purrr mapping functions. It doesn't seem to be picking up the usage when mapping across functions:

```
get_stats <- function() {
  v <- 1:10
  funs <- list(
    mean,
    median,
    mode
  )
  out <- purrr::map(funs, ~.(v = v))
  return(out)
}
```
Is giving me a "local variable `v` defined but not used linting error

**Comment by stufield** [comment]
Confirmed, I've seen this in a few of our internal packages, but it seems even the `tidyverse` package is not immune ;)

https://github.com/tidyverse/tidyverse/blob/1d7f9b71dc7e5c24dbfa1d5a83dbb7411bb66703/R/conflicts.R#L22

**Comment by AshesITR** [comment]
The problem is that tecnically, codetools is correct, because `~.(v = v)` is a formula that doesn't use `v`.
You can rewrite this using base R function syntax and it will not lint:

```r
get_stats <- function() {
  v <- 1:10
  funs <- list(
    mean,
    median,
    mode
  )
  out <- purrr::map(funs, function(f) f(v = v))
  return(out)
}

#  or,  using lambda notation

get_stats <- function() {
  v <- 1:10
  funs <- list(
    mean,
    median,
    mode
  )
  out <- purrr::map(funs, \(f) f(v = v))
  return(out)
}
```
